client/my-sites/stats/stats-chart-tabs/chart-header.jsx

```diff
@@ -1,34 +1,65 @@
+import { recordTracksEvent } from '@automattic/calypso-analytics';
 import PropTypes from 'prop-types';
+import { useDispatch } from 'react-redux';
 import Legend from 'calypso/components/chart/legend';
+import IntervalDropdown from 'calypso/components/stats-interval-dropdown';
+import { useSelector } from 'calypso/state';
+import { isJetpackSite } from 'calypso/state/sites/selectors';
+import { toggleUpsellModal } from 'calypso/state/stats/paid-stats-upsell/actions';
+import useIntervals from '../hooks/use-intervals';
 
 const ChartHeader = ( {
+	siteId,
+	slug,
+	period,
+	queryParams,
 	activeTab,
-	showLegend,
 	activeLegend,
 	availableLegend,
 	onLegendClick,
 	charts,
 } ) => {
+	const intervals = useIntervals( siteId );
+	const dispatch = useDispatch();
+	const isSiteJetpackNotAtomic = useSelector( ( state ) => {
+		return isJetpackSite( state, siteId, {
+			treatAtomicAsJetpackSite: false,
+		} );
+	} );
+	const onGatedHandler = ( events, source, statType ) => {
+		// Stop the popup from showing for Jetpack sites.
+		if ( isSiteJetpackNotAtomic ) {
+			return;
+		}
+
+		events.forEach( ( event ) => recordTracksEvent( event.name, event.params ) );
+		dispatch( toggleUpsellModal( siteId, statType ) );
+	};
+
 	return (
 		<div className="stats-chart-tabs__header">
 			<div className="stats-chart-tabs__header-title">{ activeTab?.label }</div>
-			{ showLegend && (
-				<Legend
-					activeCharts={ activeLegend }
-					activeTab={ activeTab }
-					availableCharts={ availableLegend }
-					clickHandler={ onLegendClick }
-					tabs={ charts }
-				/>
-			) }
+			<Legend
+				activeCharts={ activeLegend }
+				activeTab={ activeTab }
+				availableCharts={ availableLegend }
+				clickHandler={ onLegendClick }
+				tabs={ charts }
+			/>
+			<IntervalDropdown
+				slug={ slug }
+				period={ period }
+				queryParams={ queryParams }
+				intervals={ intervals }
+				onGatedHandler={ onGatedHandler }
```

Comment:
I guess we need a `onSelection` handler connected back to the original logic, but I'm happy for it to be addressed in follow up PRs.